### DISCUSSION TITLE
[Bug]: FSLeyesGUI v6.0.7.4 crash, whereas v6.0.3 works

### DISCUSSION THREAD

**Discussion by garner-code** [neurodesk/neurodesk.github.io-hugo#398] [Bug Report]
### Version

2024-01-12

### What platform are you seeing the problem on?

Local PC with port forwarding to a second local PC

### What happened

I go Neurodesk -> functional imaging -> fsl -> fsleyesGUI 6.0.7.4. The bash window opens and then disappears or minimizes to a small black segment on the neurodesktop task bar. Clicking on the minimized segment opens the task manager.

If I go Neurodesk -> functional imaging -> fsl -> fsl-6.0.7.4 and type the command fsleyes I get the following output:

fsl-6.0.7.4:~$ fsleyes
Traceback (most recent call last):
  File "/opt/fsl-6.0.7.4/bin/fsleyes", line 6, in <module>
    from fsleyes.filtermain import main
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/fsleyes/__init__.py", line 381, in <module>
    from fsleyes.main import embed, shutdown  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/fsleyes/main.py", line 46, in <module>
    import fsleyes.cliserver  as cliserver
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/fsleyes/cliserver.py", line 37, in <module>
    import fsleyes.actions.applycommandline as applycli
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/fsleyes/actions/applycommandline.py", line 28, in <module>
    import fsleyes.parseargs            as parseargs
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/fsleyes/parseargs.py", line 273, in <module>
    import fsleyes.plugins        as plugins
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/fsleyes/plugins/__init__.py", line 241, in <module>
    import fsleyes.views.canvaspanel     as canvaspanel
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/fsleyes/views/canvaspanel.py", line 20, in <module>
    from . import                    colourbarpanel
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/fsleyes/views/colourbarpanel.py", line 15, in <module>
    import fsleyes.gl.wxglcolourbarcanvas as cbarcanvas
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/fsleyes/gl/wxglcolourbarcanvas.py", line 14, in <module>
    import wx.glcanvas as wxgl
  File "/opt/fsl-6.0.7.4/lib/python3.11/site-packages/wx/glcanvas.py", line 10, in <module>
    from ._glcanvas import *
ImportError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /.singularity.d/libs/libGLdispatch.so.0)

If I go Neurodesk -> functional imaging -> fsl -> fsleyesGUI 6.0.3 then FSLeyes opens fine.

### How to reproduce

As above.

**Comment by stebo85** [comment]
Dear @kel-github,

Thank you for finding this. Are you running this by any chance on a MacBook with Apple Silicon?

Thank you
Steffen

**Reply by stebo85** [reply]
I currently cannot reproduce the bug yet. Can you reproduce it here? https://play-sydney.neurodesk.org/v2/gh/neurodesk/neurodesktop/main

**Comment by garner-code** [comment]
Hey Steffen :)

I cannot reproduce it at that location  :(
I am running the docker image on a linux PC; x86_64 6.5.0-14-generic

I have deleted the instance and the home folder, and have rerun using the
command found in 1.a) here
<https://www.neurodesk.org/docs/getting-started/neurodesktop/linux/> - but
that did not help.

Is there anything else required to completely reset my environment to be
'clean'? / is there anything else you can think of that I should try?

Also, cc'ing Chris as he is helping me troubleshoot :)

Best,
Kel


On Mon, 29 Jan 2024 at 11:48, Steffen Bollmann ***@***.***>
wrote:

> I currently cannot reproduce the bug yet. Can you reproduce it here?
> https://play-sydney.neurodesk.org/v2/gh/neurodesk/neurodesktop/main
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/NeuroDesk/discussions/398#discussioncomment-8275635>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABXC344SUVS4AYP5XRU4RDTYQ3WXTAVCNFSM6AAAAABCOSI6E2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DENZVGYZTK>
> .
> You are receiving this because you were mentioned.Message ID:
> <NeuroDesk/neurodesk.github.io/repo-discussions/398/comments/8275635@
> github.com>
>


-- 
Kelly Garner, PhD, she/her
Lecturer in Computational Cognitive Science
School of Psychology
UNSW

Google Scholar
<https://scholar.google.co.uk/citations?user=nmeSlZkAAAAJ&hl=en>

--I acknowledge the Traditional Owners of the land upon which I live and
work. I pay my respects to Elders past, present and emerging.--

--I seek to maintain a good work-life balance. If you receive this email
outside office hours then don’t feel you have to respond. Please expect the
like from me.--

**Comment by stebo85** [comment]
Dear Kelly,

That’s indeed very strange. Would you have time for a quick debug session Wednesday afternoon after 1pm Brisbane time?

Can you send more details in the meantime about the machine this container is running on. What linux distribution is this? How did you install docker and which version is this at?

Thank you
Steffen

From: Kelly ***@***.***>
Date: Monday, 29 January 2024 at 1:04 pm
To: NeuroDesk/neurodesk.github.io ***@***.***>
Cc: Steffen Bollmann ***@***.***>, Comment ***@***.***>
Subject: Re: [NeuroDesk/neurodesk.github.io] [Bug]: FSLeyesGUI v6.0.7.4 crash, whereas v6.0.3 works (Discussion #398)
Hey Steffen :)

I cannot reproduce it at that location :(
I am running the docker image on a linux PC; x86_64 6.5.0-14-generic

I have deleted the instance and the home folder, and have rerun using the
command found in 1.a) here
<https://www.neurodesk.org/docs/getting-started/neurodesktop/linux/> - but
that did not help.

Is there anything else required to completely reset my environment to be
'clean'? / is there anything else you can think of that I should try?

Also, cc'ing Chris as he is helping me troubleshoot :)

Best,
Kel


On Mon, 29 Jan 2024 at 11:48, Steffen Bollmann ***@***.***>
wrote:

> I currently cannot reproduce the bug yet. Can you reproduce it here?
> https://play-sydney.neurodesk.org/v2/gh/neurodesk/neurodesktop/main
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/NeuroDesk/discussions/398#discussioncomment-8275635>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABXC344SUVS4AYP5XRU4RDTYQ3WXTAVCNFSM6AAAAABCOSI6E2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DENZVGYZTK>
> .
> You are receiving this because you were mentioned.Message ID:
> <NeuroDesk/neurodesk.github.io/repo-discussions/398/comments/8275635@
> github.com>
>


--
Kelly Garner, PhD, she/her
Lecturer in Computational Cognitive Science
School of Psychology
UNSW

Google Scholar
<https://scholar.google.co.uk/citations?user=nmeSlZkAAAAJ&hl=en>

--I acknowledge the Traditional Owners of the land upon which I live and
work. I pay my respects to Elders past, present and emerging.--

--I seek to maintain a good work-life balance. If you receive this email
outside office hours then don’t feel you have to respond. Please expect the
like from me.--

**Comment by garner-code** [comment]
Hi Steffen,

I would love a debug session! I can't do Wednesday or Thursday though :( I
can do Friday any time after midday, and pretty much any time next week.

I am running the docker image on a linux PC with the following specs:
kernel version: 14~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon Nov 20 18:15:30
UTC 2
release: 6.5.0-14-generic
architecture: x86_64

I can also give you the details of the machine I am port forwarding to, if
that could help?

Chris installed docker following this:
https://docs.docker.com/engine/install/ubuntu/ I am using version 25.0.0,
build e758fe5

Let me know if I can provide any further intel :)
Kel


On Mon, 29 Jan 2024 at 14:51, Steffen Bollmann ***@***.***>
wrote:

> Dear Kelly,
>
> That’s indeed very strange. Would you have time for a quick debug session
> Wednesday afternoon after 1pm Brisbane time?
>
> Can you send more details in the meantime about the machine this container
> is running on. What linux distribution is this? How did you install docker
> and which version is this at?
>
> Thank you
> Steffen
>
> From: Kelly ***@***.***>
> Date: Monday, 29 January 2024 at 1:04 pm
> To: NeuroDesk/neurodesk.github.io ***@***.***>
> Cc: Steffen Bollmann ***@***.***>, Comment ***@***.***>
> Subject: Re: [NeuroDesk/neurodesk.github.io] [Bug]: FSLeyesGUI v6.0.7.4
> crash, whereas v6.0.3 works (Discussion #398)
> Hey Steffen :)
>
> I cannot reproduce it at that location :(
> I am running the docker image on a linux PC; x86_64 6.5.0-14-generic
>
> I have deleted the instance and the home folder, and have rerun using the
> command found in 1.a) here
> <https://www.neurodesk.org/docs/getting-started/neurodesktop/linux/> -
> but
> that did not help.
>
> Is there anything else required to completely reset my environment to be
> 'clean'? / is there anything else you can think of that I should try?
>
> Also, cc'ing Chris as he is helping me troubleshoot :)
>
> Best,
> Kel
>
>
> On Mon, 29 Jan 2024 at 11:48, Steffen Bollmann ***@***.***>
> wrote:
>
> > I currently cannot reproduce the bug yet. Can you reproduce it here?
> > https://play-sydney.neurodesk.org/v2/gh/neurodesk/neurodesktop/main
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/orgs/NeuroDesk/discussions/398#discussioncomment-8275635>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ABXC344SUVS4AYP5XRU4RDTYQ3WXTAVCNFSM6AAAAABCOSI6E2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DENZVGYZTK>
>
> > .
> > You are receiving this because you were mentioned.Message ID:
> > <NeuroDesk/neurodesk.github.io/repo-discussions/398/comments/8275635@
> > github.com>
> >
>
>
> --
> Kelly Garner, PhD, she/her
> Lecturer in Computational Cognitive Science
> School of Psychology
> UNSW
>
> Google Scholar
> <https://scholar.google.co.uk/citations?user=nmeSlZkAAAAJ&hl=en>
>
> --I acknowledge the Traditional Owners of the land upon which I live and
> work. I pay my respects to Elders past, present and emerging.--
>
> --I seek to maintain a good work-life balance. If you receive this email
> outside office hours then don’t feel you have to respond. Please expect
> the
> like from me.--
>
> —
> Reply to this email directly, view it on GitHub<
> https://github.com/orgs/NeuroDesk/discussions/398#discussioncomment-8276378>,
> or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/AA6V2W23RE7QAXXVEOJPXULYQ4GUBAVCNFSM6AAAAABCOSI6E2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DENZWGM3TQ>.
>
> You are receiving this because you commented.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/NeuroDesk/discussions/398#discussioncomment-8276635>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABXC344QTQDCUPO7ZHCMYJ3YQ4MDHAVCNFSM6AAAAABCOSI6E2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DENZWGYZTK>
> .
> You are receiving this because you were mentioned.Message ID:
> <NeuroDesk/neurodesk.github.io/repo-discussions/398/comments/8276635@
> github.com>
>


-- 
Kelly Garner, PhD, she/her
Lecturer in Computational Cognitive Science
School of Psychology
UNSW

Google Scholar
<https://scholar.google.co.uk/citations?user=nmeSlZkAAAAJ&hl=en>

--I acknowledge the Traditional Owners of the land upon which I live and
work. I pay my respects to Elders past, present and emerging.--

--I seek to maintain a good work-life balance. If you receive this email
outside office hours then don’t feel you have to respond. Please expect the
like from me.--

**Reply by stebo85** [reply]
Ok, that's helpful. I cannot reproduce it in Ubutnu 22.04 - so I guess there must be something else going on. Friday is packed for me unfortunately, would tomorrow Tuesday 10am Brisbane time or 2.30pm work?

**Comment by garner-code** [comment]
I could do tomorrow at 3pm, if that works? If not, next week I am free as a
bird :)

On Mon, 29 Jan 2024 at 16:57, Steffen Bollmann ***@***.***>
wrote:

> Ok, that's helpful. I cannot reproduce it in Ubutnu 22.04 - so I guess
> there must be something else going on. Friday is packed for me
> unfortunately, would tomorrow Tuesday 10am Brisbane time or 2.30pm work?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/NeuroDesk/discussions/398#discussioncomment-8277249>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABXC343WPMKVLYX6YWBYF43YQ424NAVCNFSM6AAAAABCOSI6E2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DENZXGI2DS>
> .
> You are receiving this because you were mentioned.Message ID:
> <NeuroDesk/neurodesk.github.io/repo-discussions/398/comments/8277249@
> github.com>
>


-- 
Kelly Garner, PhD, she/her
Lecturer in Computational Cognitive Science
School of Psychology
UNSW

Google Scholar
<https://scholar.google.co.uk/citations?user=nmeSlZkAAAAJ&hl=en>

--I acknowledge the Traditional Owners of the land upon which I live and
work. I pay my respects to Elders past, present and emerging.--

--I seek to maintain a good work-life balance. If you receive this email
outside office hours then don’t feel you have to respond. Please expect the
like from me.--

**Reply by stebo85** [reply]
Ok, I can try to shuffle meetings around to make this work. This is 3pm Brisbane time, so 4pm Sydney time?

**Comment by garner-code** [comment]
Oh, don't shuffle! I can make your 2:30pm :)
*can't encode time differences

On Mon, 29 Jan 2024 at 17:29, Steffen Bollmann ***@***.***>
wrote:

> Ok, I can try to shuffle meetings around to make this work. This is 3pm
> Brisbane time, so 4pm Sydney time?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/NeuroDesk/discussions/398#discussioncomment-8277449>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABXC34ZIRX3DOLSZWSIITO3YQ46VXAVCNFSM6AAAAABCOSI6E2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DENZXGQ2DS>
> .
> You are receiving this because you were mentioned.Message ID:
> <NeuroDesk/neurodesk.github.io/repo-discussions/398/comments/8277449@
> github.com>
>


-- 
Kelly Garner, PhD, she/her
Lecturer in Computational Cognitive Science
School of Psychology
UNSW

Google Scholar
<https://scholar.google.co.uk/citations?user=nmeSlZkAAAAJ&hl=en>

--I acknowledge the Traditional Owners of the land upon which I live and
work. I pay my respects to Elders past, present and emerging.--

--I seek to maintain a good work-life balance. If you receive this email
outside office hours then don’t feel you have to respond. Please expect the
like from me.--
